mlflow/keras/save.py

```diff
@@ -0,0 +1,341 @@
+import importlib
+import logging
+import os
+import shutil
+import tempfile
+
+import keras
+import yaml
+
+import mlflow
+from mlflow import pyfunc
+from mlflow.exceptions import INVALID_PARAMETER_VALUE, MlflowException
+from mlflow.models import (
+    Model,
+    ModelInputExample,
+    ModelSignature,
+    infer_pip_requirements,
+)
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.models.utils import _save_example
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.types.schema import TensorSpec
+from mlflow.utils.annotations import experimental
+from mlflow.utils.docstring_utils import LOG_MODEL_PARAM_DOCS, format_docstring
+from mlflow.utils.environment import (
+    _CONDA_ENV_FILE_NAME,
+    _CONSTRAINTS_FILE_NAME,
+    _PYTHON_ENV_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+    _mlflow_conda_env,
+    _process_conda_env,
+    _process_pip_requirements,
+    _PythonEnv,
+)
+from mlflow.utils.file_utils import get_total_file_size, write_to
+from mlflow.utils.requirements_utils import _get_pinned_requirement
+
+FLAVOR_NAME = "keras"
+
+_MODEL_SAVE_PATH = "model"
+_KERAS_MODULE_SPEC_PATH = "keras_module.txt"
+
+_logger = logging.getLogger(__name__)
+
+
+def get_default_pip_requirements():
+    """
+    Returns:
+        A list of default pip requirements for MLflow Models produced by Keras flavor. Calls to
+        `save_model()` and `log_model()` produce a pip environment that, at minimum, contains these
+        requirements.
+    """
+    return [_get_pinned_requirement("keras")]
```

Comment:
Is there a way to detect the backend framework (TF, torch, Jax)?

Reply:
Very good point, let me add the backend framework information.

Reply:
@chenmoneygithub can you add it?

Reply:
From Haru: let's use this utility function to retrieve the backend package + version.

Reply:
I tested and verified without any extra code, `_get_pinned_requirement` already retrieves the backend requirement, idk why tho.